sysdeps/managarm/generic/sched.cpp

```diff
@@ -0,0 +1,96 @@
+#include <bits/ensure.h>
+#include <unistd.h>
+
+#include <hel.h>
+#include <hel-syscalls.h>
+#include <mlibc/debug.hpp>
+#include <mlibc/allocator.hpp>
+#include <mlibc/posix-pipe.hpp>
+#include <mlibc/posix-sysdeps.hpp>
+
+#include <posix.frigg_bragi.hpp>
+
+namespace mlibc {
+
+int sys_getaffinity(pid_t pid, size_t cpusetsize, cpu_set_t *mask) {
+	return sys_getthreadaffinity(pid, cpusetsize, mask);
+}
+
+int sys_getthreadaffinity(pid_t tid, size_t cpusetsize, cpu_set_t *mask) {
+	SignalGuard sguard;
+
+	managarm::posix::GetAffinityRequest<MemoryAllocator> req(getSysdepsAllocator());
+
+	req.set_pid(tid);
+
+	auto [offer, send_head, recv_resp, recv_data] =
+		exchangeMsgsSync(
+			getPosixLane(),
+			helix_ng::offer(
+				helix_ng::sendBragiHeadOnly(req, getSysdepsAllocator()),
+				helix_ng::recvInline(),
+				helix_ng::recvBuffer(mask, cpusetsize)
+			)
+		);
+
+	HEL_CHECK(offer.error());
+	HEL_CHECK(send_head.error());
+	HEL_CHECK(recv_resp.error());
```

Comment:
Missing check for the `recv_data` item. Also you're not setting `req.size` which the other side expects.

Reply:
There is a check for `recv_data` on line 50.

Reply:
Oh oops.